ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -55,13 +94,299 @@ trait StorageBackend[DB_BATCH] {
+  def ledgerEndOffsetAndSequentialId(connection: Connection): (Offset, Long)
+  def initialLedgerEnd(connection: Connection): Option[Offset]
+
+  /** Part of pruning process, this needs to be in the same transaction as the other pruning related database operations
+    */
+  def updatePrunedUptoInclusive(prunedUpToInclusive: Offset)(connection: Connection): Unit
+  def prunedUptoInclusive(connection: Connection): Option[Offset]
+}
+
+trait ConfigurationStorageBackend {
+  def ledgerConfiguration(connection: Connection): Option[(Offset, Configuration)]
+  def configurationEntries(
+      startExclusive: Offset,
+      endInclusive: Offset,
+      pageSize: Int,
+      queryOffset: Long,
+  )(connection: Connection): Vector[(Offset, ConfigurationEntry)]
+}
+
+trait PartyStorageBackend {
+  def partyEntries(
+      startExclusive: Offset,
+      endInclusive: Offset,
+      pageSize: Int,
+      queryOffset: Long,
+  )(connection: Connection): Vector[(Offset, PartyLedgerEntry)]
+  def parties(parties: Seq[Ref.Party])(connection: Connection): List[PartyDetails]
+  def knownParties(connection: Connection): List[PartyDetails]
+}
+
+trait PackageStorageBackend {
+  def lfPackages(connection: Connection): Map[PackageId, PackageDetails]
+  def lfArchive(packageId: PackageId)(connection: Connection): Option[Array[Byte]]
+  def packageEntries(
+      startExclusive: Offset,
+      endInclusive: Offset,
+      pageSize: Int,
+      queryOffset: Long,
+  )(connection: Connection): Vector[(Offset, PackageLedgerEntry)]
+}
+
+trait DeduplicationStorageBackend {
+  def deduplicatedUntil(deduplicationKey: String)(connection: Connection): Instant
+  def upsertDeduplicationEntry(
+      key: String,
+      submittedAt: Instant,
+      deduplicateUntil: Instant,
+  )(connection: Connection): Int
+  def removeExpiredDeduplicationData(currentTime: Instant)(connection: Connection): Unit
+  def stopDeduplicatingCommand(deduplicationKey: String)(connection: Connection): Unit
+}
+
+trait CompletionStorageBackend {
+  def commandCompletions(
+      startExclusive: Offset,
+      endInclusive: Offset,
+      applicationId: ApplicationId,
+      parties: Set[Ref.Party],
+  )(connection: Connection): List[CompletionStreamResponse]
+
+  /** Part of pruning process, this needs to be in the same transaction as the other pruning related database operations
+    */
+  def pruneCompletions(pruneUpToInclusive: Offset)(connection: Connection): Unit
+}
+
+trait ContractStorageBackend {
+  def contractKeyGlobally(key: Key)(connection: Connection): Option[ContractId]
+  def maximumLedgerTime(ids: Set[ContractId])(connection: Connection): Try[Option[Instant]]
+  def keyState(key: Key, validAt: Long)(connection: Connection): KeyState
+  def contractState(contractId: ContractId, before: Long)(
+      connection: Connection
+  ): Option[StorageBackend.RawContractState]
+  def activeContractAndLoadArgument(readers: Set[Ref.Party], contractId: ContractId)(
```

Comment:
this one is a a weird mixture without the `get` prefix

Reply:
good point, fixing it